sstables/sstable_set.cc

```diff
@@ -1381,7 +1382,7 @@ sstable_set::make_local_shard_sstable_reader(
         }
         return reader;
     };
-    if (_impl->size() == 1) [[unlikely]] {
+    if (_impl->size() == 1 && !statistics) {
         auto sstables = _impl->all();
```

Comment:
ditto.

Reply:
also, with a single reader, all rows will come from a single sstable. I suggest we leave the stats unfilled like today if compaction works with a single sstable.

Reply:
so one can interpret it as no merging across sstables happened.

Reply:
> we cannot afford to lose the optimization, the stats is less important, see: [f75154a](https://github.com/scylladb/scylladb/commit/f75154afca472251391445697202d980a510c9a5)

Right. Definitely sounds reasonable to keep the optimization. Then, as you proposed, I will leave the field empty.